compiler/natives/src/reflect/reflect_test.go

```diff
@@ -59,40 +59,20 @@ func TestSelectOnInvalid(t *testing.T) {
 	})
 }
 
-func TestStructOfFieldName(t *testing.T) {
-	t.Skip("StructOf")
-}
-
-func TestStructOf(t *testing.T) {
-	t.Skip("StructOf")
-}
-
-func TestStructOfExportRules(t *testing.T) {
-	t.Skip("StructOf")
-}
-
-func TestStructOfGC(t *testing.T) {
-	t.Skip("StructOf")
-}
-
-func TestStructOfAlg(t *testing.T) {
-	t.Skip("StructOf")
-}
-
-func TestStructOfGenericAlg(t *testing.T) {
-	t.Skip("StructOf")
-}
-
 func TestStructOfDirectIface(t *testing.T) {
-	t.Skip("StructOf")
+	t.Skip("reflect.Value.InterfaceData is not supported by GopherJS.")
 }
 
 func TestStructOfWithInterface(t *testing.T) {
-	t.Skip("StructOf")
-}
-
-func TestStructOfTooManyFields(t *testing.T) {
-	t.Skip("StructOf")
+	// TODO(nevkontakte) Most of this test actually passes, but there is something
+	// about embedding fields with methods that can or can't be stored in an
+	// interface value directly that GopherJS does differently from upstream. As
+	// a result, GopherJS's implementation of StructOf() doesn't panic where
+	// upstream does. It seems to be a result of our implementation not propagating
+	// the kindDirectIface flag in struct types created by StructOf(), but at this
+	// point I wasn't able to figure out what that flag actually means in the
+	// GopherJS context or how it maps onto our own reflection implementation.
```

Comment:
Are these types of TODOs being documented somewhere, other than these comments? Should they be? Maybe a GH issue?

Reply:
I have thought about this a bit, but did not come to a definitive conclusion. My thoughts so far:

 - Documenting this kind of problems in the code as a comment/todo makes sense because that's where people would be looking when debugging some kind of unexpected behavior. So a TODO is generally better discoverable.
 - Making such comments with a TODO makes it easy to find them with a simple grep command. Adding a username also points to a person who might be able to provide more context for the comment if the need arises.
 - In this particular instance the problem is fuzzy enough and seemingly unimpactful, so a GitHub issue is likely to linger unfixed ~forever. This will decrease signal/noise ratio and make issues harder to manager.
 
At least for now I lean towards filing issues for things that a definitely worth fixing or likely to impact users, but I can be convinced otherwise :)